From: Martin Michlmayr <tbm@cyrius.com>
Date: Sat, 11 Apr 2009 17:05:35 +0200
Subject: [PATCH] Fix attempt to free a non-heap object

Spotted by Martin Michlmayr thanks to a new warning in GCC 4.4.
See Debian bug #505045.

Signed-off-by: Martin Michlmayr <tbm@cyrius.com>
Signed-off-by: Stephane Glondu <steph@glondu.net>
---
 src/libmojave-external/cutil/lm_printf.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/libmojave-external/cutil/lm_printf.c b/src/libmojave-external/cutil/lm_printf.c
index dc14927..5d9075d 100644
--- a/src/libmojave-external/cutil/lm_printf.c
+++ b/src/libmojave-external/cutil/lm_printf.c
@@ -142,12 +142,12 @@ value ml_print_string(value v_fmt, value v_string)
 #endif
     if(code < 0) {
         if(bufp != buffer)
-            free(buffer);
+            free(bufp);
         failwith("ml_print_string");
     }
     v_result = copy_string(bufp);
     if(bufp != buffer)
-        free(buffer);
+        free(bufp);
     return v_result;
 }
 
@@ -190,12 +190,12 @@ value ml_print_string2(value v_width, value v_fmt, value v_string)
 #endif
     if(code < 0) {
         if(bufp != buffer)
-            free(buffer);
+            free(bufp);
         failwith("ml_print_string");
     }
     v_result = copy_string(bufp);
     if(bufp != buffer)
-        free(buffer);
+        free(bufp);
     return v_result;
 }
 
-- 
